diff --git a/apiserver/plane/settings/production.py b/apiserver/plane/settings/production.py
index 806f83aca..ea60e5b19 100644
--- a/apiserver/plane/settings/production.py
+++ b/apiserver/plane/settings/production.py
@@ -46,9 +46,9 @@ LOGGING = {
         "file": {
             "class": "plane.utils.logging.SizedTimedRotatingFileHandler",
             "filename": (
-                os.path.join(BASE_DIR, "logs", "plane-debug.log")  # noqa
+                os.path.join(os.environ.get("RUNTIME_DIR", BASE_DIR), "logs", "plane-debug.log")  # noqa
                 if DEBUG
-                else os.path.join(BASE_DIR, "logs", "plane-error.log")  # noqa
+                else os.path.join(os.environ.get("RUNTIME_DIR", BASE_DIR), "logs", "plane-error.log")  # noqa
             ),
             "when": "s",
             "maxBytes": 1024 * 1024 * 1,
